sklearn/utils/validation.py

```diff
@@ -1655,6 +1655,51 @@ def check_is_fitted(estimator, attributes=None, *, msg=None, all_or_any=all):
         raise NotFittedError(msg % {"name": type(estimator).__name__})
 
 
+def _estimator_has(attr, delegates=None):
+    """Check if we can delegate a method to the underlying estimator.
+
+    We check the `delegates` in the order they are passed.
+    By default, we first check the fitted estimator if available,
+    otherwise we check the unfitted estimator.
+
+    getattr() returns True if the attribute exists in the object,
```

Comment:
Here this is a bit weird to speak about `getattr` because it forces us to read the code base. The fact that `getattr` is rather an implementation details.

I think that we can remove this sentence and instead, add in the "Raises" section that an `AttributeError` is raised if the attribute does not exist in the object.

We can add the remaining details in the "Returns" for the `check` argument mentioning that the callable returns True if the attribute exist otherwise it raise an `AttributeError`.